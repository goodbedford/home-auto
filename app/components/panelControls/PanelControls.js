import React from "react";
import {NavLink} from "react-router-dom";

const PropTypes = React.PropTypes;
const propTypes = {
  header: PropTypes.string.isRequired
};

const PanelControls = (props) => {

  return (
    <div className="controls controls-flex controls-flex--center-start">
      <div className="controls__group controls__group--space-between controls__group--tall">
        <NavLink to={`/homes/${props.match.params.homeId}/rooms/new`}>
          <span className="controls__icon controls__icon--sm invisible">+</span>
        </NavLink>
        <span className="controls__text controls__text-btn">{props.header}</span>
        {/* room control btns */}
        {this.children}
        <div className="block-tall">
          <span className="block-controls"></span>
        </div>
      </div>
    </div>
  );
};
PanelControls.propTypes = propTypes;
export default PanelControls;
