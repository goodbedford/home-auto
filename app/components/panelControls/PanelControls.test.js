/* eslint-disable no-undef */

import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import PanelControls from "./PanelControls";

describe("<PanelControls />", () => {
  const props = {
    header: "rooms",
    match: {params: {homeId: 12}}
  };
  it("PanelControls snapshot", () => {
    const wrapper = shallow(
      <PanelControls
        {...props}
      />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
