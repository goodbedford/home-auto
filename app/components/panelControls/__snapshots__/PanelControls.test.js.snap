// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PanelControls /> PanelControls snapshot 1`] = `
<div
  className="controls controls-flex controls-flex--center-start"
>
  <div
    className="controls__group controls__group--tall controls__group--space-between"
  >
    <NavLink
      activeClassName="active"
      to="/homes/12/rooms/new"
    >
      <span
        className="controls__icon controls__icon--sm invisible"
      >
        +
      </span>
    </NavLink>
    <span
      className="controls__text controls__text-btn"
    >
      rooms
    </span>
    <div
      className="block-tall"
    >
      <span
        className="block-controls"
      />
    </div>
  </div>
</div>
`;
